source/channel_bootstrap.c

```diff
@@ -635,8 +648,7 @@ static void s_on_host_resolved(
     if (err_code == 0) {
```

Comment:
commenting on bug I found a bunch of lines up:
```
        if (client_connection_args->failed_count < client_connection_args->addresses_count) {
            s_connection_args_release(client_connection_args);
            return;
        }
```

This for loop will starts a socket connection for each address. After the for-loop, unless EVERY socket failed, we release a refcount on the connection args and assume things will resume when the socket connect callbacks fire.

BUT BUT BUT if during the for-loop we fail to allocator or init a socket, we break out of the loop. The logic was probably "well, if this failed then the rest surely will also fail". But that breaks the `failed_count < addresses_count` check later. If the 1st socket failed, then there are no pending socket connects out there

Reply:
probably solution is incrementing client_connection_args->addresses_count++ at the start of each for-loop, instead of setting it once before going into the for-loop.

Reply:
This turned out to be much harder to fix than it looks. We will need to re-visit this when on_host_resolved takes place in an event loop.

Reply:
I ended up actually fixing this, by moving the socket creation into the event loop